### PR TITLE
DOCSUP-11556: Document the Pad functions

### PR DESCRIPTION
**Pull request by ka1bi4**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Documentation for #26075

Changelog entry:
- Add new functions `leftPad()`, `rightPad()`, `leftPadUTF8()`, `rightPadUTF8()`.

### CODE REVIEW

File: docs/en/sql-reference/functions/string-functions.md [hunk 1]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
 The result type is UInt64.
 
+## leftPad {#leftpad}
+
+Pads the current string from the left with a specified string (multiple times, if needed) until the resulting string reaches the given length. Similarly to the MySQL `LPAD` function.
+
+**Syntax**
+
+``` sql
+leftPad('string','length', 'pad_string')
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:53]:
```suggestion
leftPad('string', 'length', 'pad_string')
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 2]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+```
+
+**Arguments**
+
+-   `string` — Input string, that need to be padded. [String](../data-types/string.md).
+-   `length` — The length of the resulting string once the input string pads. [UInt](../data-types/int-uint.md). If the value is less than input string length, then string is returned as-is.
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:59]:
```suggestion
-   `length` — The length of the resulting string. [UInt](../data-types/int-uint.md). If the value is less than the input string length, then the input string is returned as-is.
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 3]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+-   `pad_string` — The string to pad the current input string with. [String](../data-types/string.md).
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:60]:
```suggestion
-   `pad_string` — The string to pad the input string with. [String](../data-types/string.md). Optional. If not specified, then the input string is padded with spaces.
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 4]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+
+[String](../data-types/string.md)
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:62]:
```suggestion

```

File: docs/en/sql-reference/functions/string-functions.md [hunk 5]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+
+**Returned value(s)**
+
+-   The resulting string reaches the given length.
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:66]:
```suggestion
-   The resulting string with the given length.
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 6]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+
+Type: [String](../data-types/string.md).
+
+**Example**
+
+Query:
+
+``` sql
+SELECT leftPad('abc', 7, '*');
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:75]:
```suggestion
SELECT leftPad('abc', 7, '*'), leftPad('def', 7);
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 7]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+```
+
+Result:
+
+``` text
+┌─leftPad('abc', 7, '*')─┐
+│ ****abc                │
+└────────────────────────┘
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:83]:
```suggestion
┌─leftPad('abc', 7, '*')─┬─leftPad('def', 7)─┐
│ ****abc                │     def           │
└────────────────────────┴───────────────────┘
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 8]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+```
+
+## leftPadUTF8 {#leftpadutf8}
+
+Pads the current string from the left with a specified string (multiple times, if needed) until the resulting string reaches the given length. Similarly to the MySQL `LPAD` function.
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:88]:
```suggestion
Pads the current string from the left with a specified string (multiple times, if needed) until the resulting string reaches the given length. Similarly to the MySQL `LPAD` function. While in the [leftPad](#leftpad) function the length is measured in bytes, here in the `leftPadUTF8` function it is measured in code points.
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 9]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+
+**Syntax**
+
+``` sql
+leftPadUTF8('string','length', 'pad_string')
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:93]:
```suggestion
leftPadUTF8('string','length'[, 'pad_string'])
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 10]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+
+**Syntax**
+
+``` sql
+leftPad('string','length', 'pad_string')
+```
+
+**Arguments**
+
+-   `string` — Input string, that need to be padded. [String](../data-types/string.md).
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:58]:
```suggestion
-   `string` — Input string, that needs to be padded. [String](../data-types/string.md).
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 11]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+
+**Syntax**
+
+``` sql
+leftPadUTF8('string','length', 'pad_string')
+```
+
+**Arguments**
+
+-   `string` — Input UTF-8 string, that need to be padded. [String](../data-types/string.md).
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:98]:
```suggestion
-   `string` — Input UTF-8 string, that needs to be padded. [String](../data-types/string.md).
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 12]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+-   `length` — The length of the resulting string once the input string pads. [UInt](../data-types/int-uint.md). If the value is less than input string length, then string is returned as-is.
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:99]:
```suggestion
-   `length` — The length of the resulting string. [UInt](../data-types/int-uint.md). If the value is less than the input string length, then the input string is returned as-is.
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 13]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+-   `pad_string` — The string to pad the current input string with. [String](../data-types/string.md).
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:100]:
```suggestion
-   `pad_string` — The string to pad the current input string with. [String](../data-types/string.md). Optional. If not specified, then the input string is padded with spaces.
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 14]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+
+[String](../data-types/string.md)
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:102]:
```suggestion

```

File: docs/en/sql-reference/functions/string-functions.md [hunk 15]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+
+**Returned value(s)**
+
+-   The resulting UTF-8 string reaches the given length.
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:106]:
```suggestion
-   The resulting UTF-8 string with the given length.
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 16]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+
+Type: [String](../data-types/string.md).
+
+**Example**
+
+Query:
+
+``` sql
+SELECT leftPadUTF8('абвг', 7, '*');
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:115]:
```suggestion
SELECT leftPadUTF8('абвг', 7, '*'), leftPadUTF8('дежз', 7);
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 17]
```diff
@@ -43,6 +43,163 @@ The result type is UInt64.
...
+```
+
+Result:
+
+``` text
+┌─leftPadUTF8('абвг', 7, '*')─┐
+│ ***абвг                     │
+└─────────────────────────────┘
```
Comment (olgarev) [inline docs/en/sql-reference/functions/string-functions.md:123]:
```suggestion
┌─leftPadUTF8('абвг', 7, '*')─┬─leftPadUTF8('дежз', 7)─┐
│ ***абвг                     │    дежз                │
└─────────────────────────────┴────────────────────────┘
```

File: docs/ru/sql-reference/functions/string-functions.md [hunk 18]
```diff
@@ -39,6 +39,158 @@ toc_title: "Функции для работы со строками"
...
+
+``` text
+┌─leftPad('abc', 7, '*')─┬─leftPad('def', 7)─┐
+│ ****abc                │     def           │
+└────────────────────────┴───────────────────┘
+```
+
+## leftPadUTF8 {#leftpadutf8}
+
+Заполняет текущую строку слева пробелами или указанной строкой (несколько раз, если необходимо), пока результирующая строка не достигнет заданной длины. Соответствует MySQL функции `LPAD`. Функция [leftPad](#leftpad) измеряет длину строки в байтах, а функция `leftPadUTF8` — в кодовых точках Unicode.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/string-functions.md:82]:
```suggestion
Заполняет текущую строку слева пробелами или указанной строкой (несколько раз, если необходимо), пока результирующая строка не достигнет заданной длины. Соответствует MySQL функции `LPAD`. В отличие от функции [leftPad](#leftpad), измеряет длину строки не в байтах, а в кодовых точках Unicode.
```

File: docs/ru/sql-reference/functions/string-functions.md [hunk 19]
```diff
@@ -39,6 +39,158 @@ toc_title: "Функции для работы со строками"
...
+
+**Синтаксис**
+
+``` sql
+leftPad('string', 'length'[, 'pad_string'])
+```
+
+**Параметры**
+
+-   `string` — входная строка, которую необходимо заполнить. [String](../data-types/string.md).
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/string-functions.md:54]:
```suggestion
-   `string` — входная строка, которую необходимо дополнить. [String](../data-types/string.md).
```

File: docs/ru/sql-reference/functions/string-functions.md [hunk 20]
```diff
@@ -39,6 +39,158 @@ toc_title: "Функции для работы со строками"
...
+
+**Синтаксис**
+
+``` sql
+leftPadUTF8('string','length'[, 'pad_string'])
+```
+
+**Параметры**
+
+-   `string` — входная строка, которую необходимо заполнить. [String](../data-types/string.md).
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/string-functions.md:92]:
```suggestion
-   `string` — входная строка, которую необходимо дополнить. [String](../data-types/string.md).
```

File: docs/ru/sql-reference/functions/string-functions.md [hunk 21]
```diff
@@ -39,6 +39,158 @@ toc_title: "Функции для работы со строками"
...
+-   `length` — длина результирующей строки. [UInt](../data-types/int-uint.md). Если указанное значение меньше, чем длина входной строки, то входная строка возвращается как есть.
+-   `pad_string` — строка, используемая для заполнения входной строки. [String](../data-types/string.md). Необязательный параметр. Если не указано, то входная строка заполняется пробелами.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/string-functions.md:94]:
```suggestion
-   `pad_string` — строка, используемая для дополнения входной строки. [String](../data-types/string.md). Необязательный параметр. Если не указано, то входная строка дополняется пробелами.
```

File: docs/ru/sql-reference/functions/string-functions.md [hunk 22]
```diff
@@ -39,6 +39,158 @@ toc_title: "Функции для работы со строками"
...
+
+``` text
+┌─leftPadUTF8('абвг', 7, '*')─┬─leftPadUTF8('дежз', 7)─┐
+│ ***абвг                     │    дежз                │
+└─────────────────────────────┴────────────────────────┘
+```
+
+## rightPad {#rightpad}
+
+Заполняет текущую строку справа пробелами или указанной строкой (несколько раз, если необходимо), пока результирующая строка не достигнет заданной длины. Соответствует MySQL функции `RPAD`.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/string-functions.md:120]:
```suggestion
Дополняет текущую строку справа пробелами или указанной строкой (несколько раз, если необходимо), пока результирующая строка не достигнет заданной длины. Соответствует MySQL функции `RPAD`.
```

File: docs/ru/sql-reference/functions/string-functions.md [hunk 23]
```diff
@@ -39,6 +39,158 @@ toc_title: "Функции для работы со строками"
...
+
+**Синтаксис**
+
+``` sql
+rightPad('string', 'length'[, 'pad_string'])
+```
+
+**Параметры**
+
+-   `string` — входная строка, которую необходимо заполнить. [String](../data-types/string.md).
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/string-functions.md:130]:
```suggestion
-   `string` — входная строка, которую необходимо дополнить. [String](../data-types/string.md).
```

File: docs/ru/sql-reference/functions/string-functions.md [hunk 24]
```diff
@@ -39,6 +39,158 @@ toc_title: "Функции для работы со строками"
...
+-   `length` — длина результирующей строки. [UInt](../data-types/int-uint.md). Если указанное значение меньше, чем длина входной строки, то входная строка возвращается как есть.
+-   `pad_string` — строка, используемая для заполнения входной строки. [String](../data-types/string.md). Необязательный параметр. Если не указано, то входная строка заполняется пробелами.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/string-functions.md:132]:
```suggestion
-   `pad_string` — строка, используемая для дополнения входной строки. [String](../data-types/string.md). Необязательный параметр. Если не указано, то входная строка дополняется пробелами.
```

File: docs/ru/sql-reference/functions/string-functions.md [hunk 25]
```diff
@@ -39,6 +39,158 @@ toc_title: "Функции для работы со строками"
...
+
+``` text
+┌─rightPad('abc', 7, '*')─┬─rightPad('abc', 7)─┐
+│ abc****                 │ abc                │
+└─────────────────────────┴────────────────────┘
+```
+
+## rightPadUTF8 {#rightpadutf8}
+
+Заполняет текущую строку слева пробелами или указанной строкой (несколько раз, если необходимо), пока результирующая строка не достигнет заданной длины. Соответствует MySQL функции `RPAD`. Функция [rightPad](#rightpad) измеряет длину строки в байтах, а функция `rightPadUTF8` — в кодовых точках Unicode.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/string-functions.md:158]:
```suggestion
Дополняет текущую строку слева пробелами или указанной строкой (несколько раз, если необходимо), пока результирующая строка не достигнет заданной длины. Соответствует MySQL функции `RPAD`. В отличие от функции [rightPad](#rightpad), измеряет длину строки не в байтах, а в кодовых точках Unicode.
```

File: docs/ru/sql-reference/functions/string-functions.md [hunk 26]
```diff
@@ -39,6 +39,158 @@ toc_title: "Функции для работы со строками"
...
+
+**Синтаксис**
+
+``` sql
+rightPadUTF8('string','length'[, 'pad_string'])
+```
+
+**Параметры**
+
+-   `string` — входная строка, которую необходимо заполнить. [String](../data-types/string.md).
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/string-functions.md:168]:
```suggestion
-   `string` — входная строка, которую необходимо дополнить. [String](../data-types/string.md).
```

File: docs/ru/sql-reference/functions/string-functions.md [hunk 27]
```diff
@@ -39,6 +39,158 @@ toc_title: "Функции для работы со строками"
...
+-   `length` — длина результирующей строки. [UInt](../data-types/int-uint.md). Если указанное значение меньше, чем длина входной строки, то входная строка возвращается как есть.
+-   `pad_string` — строка, используемая для заполнения входной строки. [String](../data-types/string.md). Необязательный параметр. Если не указано, то входная строка заполняется пробелами.
```
Comment (olgarev) [inline docs/ru/sql-reference/functions/string-functions.md:170]:
```suggestion
-   `pad_string` — строка, используемая для дополнения входной строки. [String](../data-types/string.md). Необязательный параметр. Если не указано, то входная строка дополняется пробелами.
```

File: docs/ru/sql-reference/functions/string-functions.md [hunk 28]
```diff
@@ -39,6 +39,158 @@ toc_title: "Функции для работы со строками"
...
+
+**Возвращаемое значение**
+
+-   Результирующая строка заданной длины.
+
+Type: [String](../data-types/string.md).
+
+**Пример**
+
+Результат:
```
Comment (mathalex) [inline docs/ru/sql-reference/functions/string-functions.md:180]:
```suggestion
Запрос:
```

File: docs/en/sql-reference/functions/string-functions.md [hunk 29]
```diff
@@ -43,6 +43,158 @@ The result type is UInt64.
...
+
+**Syntax**
+
+``` sql
+leftPad('string', 'length'[, 'pad_string'])
+```
+
+**Arguments**
+
+-   `string` — Input string, that needs to be padded. [String](../data-types/string.md).
```
Comment (mathalex) [inline docs/en/sql-reference/functions/string-functions.md:58]:
```suggestion
-   `string` — Input string that needs to be padded. [String](../data-types/string.md).
```
A comma before 'that' is usually not needed.

File: docs/en/sql-reference/functions/string-functions.md [hunk 30]
```diff
@@ -43,6 +43,158 @@ The result type is UInt64.
...
+
+**Syntax**
+
+``` sql
+leftPadUTF8('string','length'[, 'pad_string'])
+```
+
+**Arguments**
+
+-   `string` — Input string, that needs to be padded. [String](../data-types/string.md).
```
Comment (mathalex) [inline docs/en/sql-reference/functions/string-functions.md:96]:
```suggestion
-   `string` — Input string that needs to be padded. [String](../data-types/string.md).
```
A comma before 'that' is usually not needed.

File: docs/en/sql-reference/functions/string-functions.md [hunk 31]
```diff
@@ -43,6 +43,158 @@ The result type is UInt64.
...
+
+**Syntax**
+
+``` sql
+rightPad('string', 'length'[, 'pad_string'])
+```
+
+**Arguments**
+
+-   `string` — Input string, that needs to be padded. [String](../data-types/string.md).
```
Comment (mathalex) [inline docs/en/sql-reference/functions/string-functions.md:134]:
```suggestion
-   `string` — Input string that needs to be padded. [String](../data-types/string.md).
```
A comma before 'that' is usually not needed.

File: docs/en/sql-reference/functions/string-functions.md [hunk 32]
```diff
@@ -43,6 +43,158 @@ The result type is UInt64.
...
+
+**Syntax**
+
+``` sql
+rightPadUTF8('string','length'[, 'pad_string'])
+```
+
+**Arguments**
+
+-   `string` — Input string, that needs to be padded. [String](../data-types/string.md).
```
Comment (mathalex) [inline docs/en/sql-reference/functions/string-functions.md:172]:
```suggestion
-   `string` — Input string that needs to be padded. [String](../data-types/string.md).
```
A comma before 'that' is usually not needed.

**Review by mathalex — APPROVED** [review]

**Pull request merged**